test-suite-projectgen-micronaut/src/main/java/io/micronaut/projectgen/micronaut/features/cli/MicronautCli.java

```diff
@@ -0,0 +1,169 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.projectgen.micronaut.features.cli;
+
+import io.micronaut.core.annotation.NonNull;
+import io.micronaut.projectgen.core.buildtools.BuildTool;
+import io.micronaut.projectgen.core.buildtools.gradle.GradleDsl;
+import io.micronaut.projectgen.core.feature.Feature;
+import io.micronaut.projectgen.core.generator.GeneratorContext;
+import io.micronaut.projectgen.core.generator.ModuleContext;
+import io.micronaut.projectgen.core.options.GenericOptionsBuilder;
+import io.micronaut.projectgen.core.options.Language;
+import io.micronaut.projectgen.core.options.Options;
+import io.micronaut.projectgen.core.options.TestFramework;
+import io.micronaut.projectgen.core.template.YamlTemplate;
+import io.micronaut.projectgen.core.utils.OptionUtils;
+import jakarta.inject.Singleton;
+import org.yaml.snakeyaml.Yaml;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+@Singleton
+public class MicronautCli implements Feature {
+    private static final String KEY_APPLICATION_TYPE = "applicationType";
+    private static final String KEY_TEST_FRAMEWORK = "testFramework";
+    private static final String KEY_DEFAULT_PACKAGE = "defaultPackage";
+    private static final String KEY_SOURCE_LANGUAGE = "sourceLanguage";
+    private static final String KEY_FEATURES = "features";
+    private static final String KEY_BUILD_TOOL = "buildTool";
+    public static final String LEGACY_BUILD_TOOL_GRADLE_KOTLIN = "gradle_kotlin";
+    public static final String LEGACY_BUILD_TOOL_GRADLE_GROOVY = "gradle";
+    public static final String LEGACY_BUILD_TOOL_MAVEN = "maven";
+    public static final String PATH = "micronaut-cli.yml";
+
+    @Override
+    public String getName() {
+        return "micronaut-cli";
+    }
+
+    @Override
+    public String getTitle() {
+        return "Micronaut CLI";
+    }
+
+    @Override
+    public String getDescription() {
+        return "Adds a micronaut-cli.yml in the root of the project";
+    }
+
+    @Override
+    public void apply(GeneratorContext generatorContext) {
+        Options options = generatorContext.getOptions();
+        ModuleContext module = generatorContext.getRootModule();
+        module.addTemplate("micronautCli",
+            new YamlTemplate(PATH, config(options)));
+    }
+
+    public static Options load(File projectFolder) {
+        Map<String, Object> yamlValues = getStringObjectMap(projectFolder);
+        var builder = GenericOptionsBuilder.builder();
+        if (yamlValues.containsKey(KEY_APPLICATION_TYPE)) {
+            builder.template((String) yamlValues.get(KEY_APPLICATION_TYPE));
```

Comment:
don't force casts? Does not snakeyaml has a better utility to get a string or a list of strings?